packages/rich-text/src/plugins/Hyperlink/index.tsx

```diff
@@ -182,9 +182,7 @@ export function ToolbarHyperlinkButton(props: ToolbarHyperlinkButtonProps) {
 }
 
 const isAnchor = (element: HTMLElement) =>
-  element.nodeName === 'A' &&
-  !!element.getAttribute('href') &&
-  element.getAttribute('href') !== '#';
+  element.nodeName === 'A' && element.getAttribute('href') !== '#';
```

Comment:
This may break the URL hyperlink. Let's have a different `isAnchor` for entity links

Reply:
what about only checking for `element.getAttribute('data-link-type') === xxx`. It should be specific already, right?

Reply:
It's safer to check the HTML tag as well for entity links but we should only check `href` in case of URL hyperlinks

Reply:
Updated!